src/messages/textDocument_document.cc

```diff
@@ -165,88 +152,64 @@ void MessageHandler::textDocument_documentSymbol(JsonReader &reader,
     std::sort(result.begin(), result.end());
     reply(result);
   } else if (g_config->client.hierarchicalDocumentSymbolSupport) {
-    std::unordered_map<SymbolIdx, std::unique_ptr<DocumentSymbol>> sym2ds;
-    std::vector<std::pair<std::vector<const void *>, DocumentSymbol *>> funcs,
-        types;
+    std::vector<ExtentRef> syms;
```

Comment:
We may want this to be a vector of pointer to take less memory 32B to 8B per element and then sort the pointers.